perf/src/qos.rs

```diff
@@ -0,0 +1,121 @@
+use {
+    crate::{
+        data_budget::DataBudget,
+        packet::{Packet, PacketBatch},
+    },
+    std::{
+        collections::HashMap,
+        net::IpAddr,
+        sync::atomic::{AtomicUsize, Ordering},
+    },
+};
+
+pub type StakeMap = HashMap<IpAddr, (u64, DataBudget)>;
```

Comment:
I think you should use a 1m unit Vec<DataBudget> and ahash(ip) to the cell so we can quickly block IPs that flood with dups

Reply:
well maybe I have both. I also need to lookup if the ip is staked, no?

Reply:
yea, we do